libchisel/src/checkstartfunc.rs

```diff
@@ -84,4 +96,23 @@ mod tests {
         let result = checker.validate(&module).unwrap();
         assert_eq!(false, result);
     }
+
+    #[test]
+    fn as_abstract() {
+        let wasm: Vec<u8> = vec![
+            0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00, 0x01, 0x04, 0x01, 0x60, 0x00, 0x00,
+            0x03, 0x02, 0x01, 0x00, 0x07, 0x08, 0x01, 0x04, 0x6d, 0x61, 0x69, 0x6e, 0x00, 0x00,
+            0x0a, 0x04, 0x01, 0x02, 0x00, 0x0b,
+        ];
+
+        let module = deserialize_buffer::<Module>(&wasm).unwrap();
+
+        let checker = CheckStartFunc::new(false);
+        assert_eq!("checkstartfunc", checker.id());
+
+        let as_trait: &dyn ModuleValidator = checker.as_abstract();
```

Comment:
If you don't add these tests to the other modules, then I suggest to include this test in `lib.rs` next to the other basic module tests.